src/FindTranslocations.cpp

```diff
@@ -15,9 +16,127 @@
 
 #include "data_structures/Translocation.h"
 #include  <boost/program_options.hpp>
+#include "api/BamWriter.h"
+
 namespace po = boost::program_options;
 
 
+//returns a bam file contaning the reads located in a given region of the genome
+void TranslocationBAM(string BamFileName,string outputFileHeader,string positionData){
+	int chr[2];
+	int startPos[2];int endPos[2];
+	int arguments;
+
+	//processing the command line input
+	string buf; 
+	stringstream ss(positionData);
+	vector<string> tokens;
+	while (ss >> buf){tokens.push_back(buf);}
+	arguments=tokens.size();
+
+
+	string tmpstr;
+	if(arguments == 6){
+		try{
+			for(int i=0;i<=1;i++){
+			//extracts the chromosome REFID
+			tmpstr=tokens.front();
+			tokens.erase(tokens.begin());
+			tmpstr=tmpstr.substr(3);
+			
+
+			bool isNumber = true;
+			for(string::const_iterator k = tmpstr.begin(); k != tmpstr.end(); ++k){
+				if(isdigit(*k)==0){isNumber=0;}
+			}
+			//if the ID of the chromosome is given as a number, the refID is a number, if the X or Y chromosome is given, the correct refID must be found.
+			if(isNumber==1){
+				chr[i] = atoi(tmpstr.c_str());
```

Comment:
much better if you use chr2pos and pos2chr that I create, this solution is not general enough:
 - there are many others "chromosomes" in the human genome (unplaced contigs, unknowns, mitochondrial) 
 - there is no reason to lock the tool to the human genome, if you are general enough this tool can be used in all available genomes